src/main/java/duke/main/Parser.java

```diff
@@ -0,0 +1,126 @@
+package main.java.duke.main;
+
+import java.util.Scanner;
+
+import main.java.duke.exception.EmptyDateException;
+import main.java.duke.exception.EmptyTaskException;
+import main.java.duke.exception.InvalidDateFormatException;
+import main.java.duke.exception.InvalidIndexException;
+
+/**
+ * Reads and parses the user input.
+ */
+public class Parser {
+    private static final Scanner sc = new Scanner(System.in);
+
+    /**
+     * Reads next line of input from the user.
+     * @return The next line that user inputs.
```

Comment:
Same as above, might be good to leave an empty line between description and params/return statements.